### PR TITLE
Update to a two column UI for manual data entry screen with basic validation warnings

### PR DESCRIPTION
**Pull request by carolinemodic**
This PR does two main things
 - Iterated on the UI of the manual data page, breaking it into two columns. @beausmith would love your feedback here. The sticky behavior of the second column is lovely when the summary column is short enough to not need to scroll but when there are a lot of precincts and that side of the screen ends up scrolling the behavior is a little clunky feeling and I'm not sure how to improve it. I also wasn't sure the best location for the "select precinct" dropdown, if it should be on top of the right column with the precinct-based data or on the left sidebar. 
- Adds basic validation warnings when the contest ballots don't add up to the inputted totals, these warnings are also highlighted in the summary table. They do NOT block the ability to save the data. 

https://user-images.githubusercontent.com/14897017/115928167-8c124c80-a43a-11eb-98f4-0856165b6cb3.mov

### CODE REVIEW

**Comment by beausmith** [comment]
Reviewing this PR…

**Comment by carolinemodic** [comment]
Didn't get this today so just writing out the plan before I forget: 

Remove the two column layout, instead have index page with the precinct summarys in a table as they are now with buttons to go to edit the data for each precinct. Clicking this goes to a new screen with the contest details to fill out and a save button at the bottom, saves and takes you back to the index page. From the index page you can go back to the main Tally screen but now actual saving happens. When you toggle precinct/absentee ballots various text on the page updates (summary table "Edit Absentee Ballot Data", header on the page, etc.)

**Comment by beausmith** [comment]
@carolinemodic - Summery looks good other than one item: I think "actual saving" should happen anytime that results are saved for a specific precinct. When you go back to the Manual Tally index screen, you shouldn't need to save anything. If the user clicks a different tab in the UI they shouldn't "lose" any unsaved data. 

Perhaps the button to delete all manually saved data should be moved to the Manual Tally index screen?

**Comment by beausmith** [comment]
@carolinemodic - Please ping me when updated with new flow.

**Comment by benadida** [comment]
I'd like to leave this review to @beausmith

**Comment by eventualbuddha** [comment]
Still relevant?

**Comment by carolinemodic** [comment]
Updated to two screen format previously discussed with @beausmith 

Video: 

https://user-images.githubusercontent.com/14897017/120040166-f2026e80-bfba-11eb-9ee5-082f51ce8afb.mov

**Review by eventualbuddha — CHANGES_REQUESTED** [review]
Reviewing as "Request changes" so the Slackbot will stop pinging us about this.

**Comment by beausmith** [comment]
@carolinemodic Thanks for this update! It looks great. I'd like to make some adjustments to this PR and send it back to you for review. I'll do this today.

**Comment by beausmith** [comment]
Ok @carolinemodic… this branch is rebased on master and has a design applied. 

## TODO
- [ ] Some tests may not pass.
- [ ] Total Ballots Entered on index doesn't match ballots added on precinct screen because I removed the field for ballots entered.

**Comment by carolinemodic** [comment]
@beausmith Back to you for review on this! We just had to explictly calculate and set the total number of ballots when we're saving the tally now that that isn't an imputable field. So this updates it to always calculate and set the number of ballots in a contest in the internal tally object when other entries for that contest are modified, and the number of ballots shown on the screen is grabbed from the internal tally where that is set instead of being calculated again.

File: apps/election-manager/src/screens/ManualDataImportPrecinctScreen.tsx [hunk 1]
```diff
@@ -365,7 +364,7 @@ const ManualDataImportPrecinctScreen: React.FC = () => {
               Save Data for {currentPrecinct.name}
             </Button>
           </p>
-          <p>Enter the umber of votes for each contest option.</p>
+          <p>Enter the number of votes for each contest option.</p>
```
Comment (beausmith) [inline apps/election-manager/src/screens/ManualDataImportPrecinctScreen.tsx:367]: 😮

**Review by beausmith — APPROVED** [review]
@carolinemodic - Thank you so much for updating the logic and tests on this PR. I've made one more commit to remove the buttons at the top of the data entry screen and updated the button text at the bottom of that screen. If this commit looks good to you, merge it!!

**Pull request merged**